Solution for issue #17

mhoangvslev opened the pull request:
- Inform the users via output cell, the number of rows in the result
- If the result has 0 entry, it wont appear in Latex 
- Change the version to 0.2
- The output looks like this:
https://github.com/mhoangvslev/JupyterPostgreSQLBasics/blob/master/TP01.pdf

mhoangvslev committed:
Solution for issue #17

- Inform the users via output cell, the number of rows in the result
- If the result has 0 entry, it wont appear in Latex 
- Change the version to 0.2

bgschiller commented on postgres_kernel.egg-info/PKG-INFO:
```diff
@@ -8,7 +8,7 @@ Author-email: bgschiller@gmail.com
 Description: A simple Jupyter kernel for PostgreSQL
         
-        I don't know how to set this up to be pip-installable, but maybe you can help with that?
+        Install with `pip install git+https://github.com/bgschiller/postgres_kernel.git`
```
👍

bgschiller commented on postgres_kernel/kernel.py:
```diff
@@ -181,16 +181,24 @@ def do_execute(self, code, silent, store_history=True,
+                    self.iopub_socket, 'stream', {
+                        'name': 'stdout',
+                        'text': str(len(rows)) + " row(s) returned."
+                        })
```
This change is ensuring we always have output, even if there's no rows in the result?

The indentation is a little funky. Do you think you could change it to use 4 spaces consistently?

mhoangvslev replied:
The comments I put in is just to tell people what have been modified without making them look up the commit history. I will remove it if you so wish, nonetheless.

bgschiller commented on postgres_kernel/kernel.py:
```diff
@@ -181,16 +181,24 @@ def do_execute(self, code, silent, store_history=True,
     d = {
         'data': {
-            'text/plain': tabulate(rows, header, tablefmt='simple'),
+            'text/latex': tabulate(rows, header, tablefmt='latex_booktabs'),
```
Any reason to drop `text/plain` support? it seems like we can add `text/latex` without removing plain text as a possible content type.

mhoangvslev replied:
I tested with all sort of combination: the first must be text/latex and the second must be either latex or latex bookstab

bgschiller replied:
Gotcha. I guess what I'm saying is: Any reason we can't do this?

```suggestion
            'text/latex': tabulate(rows, header, tablefmt='latex_booktabs'),
            'text/plain': tabulate(rows, header, tablefmt='simple'),
```

mhoangvslev replied:
If the table is printed in plain and it has many attributes, it won't be scaled to fit the page. I don't see how plain text is useful here.

bgschiller replied:
@mhoangvslev I just tested your branch, and it doesn't produce output when running in the console:

![](https://trello-attachments.s3.amazonaws.com/58d428743111af1d0a20cf28/5bfdf2eb99985f018321281f/e4b00691624a411ab278511ff1a41403/capture.png)

Here is the output of the same query in the master branch:

![](https://trello-attachments.s3.amazonaws.com/58d428743111af1d0a20cf28/5bfdf30be1abcd0453e7bbd9/2f7b37a062177810d506b77eff541490/capture.png)

I suspect this difference is because your branch is missing the `text/plain` content type. You can try this for yourself by running `jupyter console --kernel postgres`

mhoangvslev replied:
Doubtlessly, I never think about console usage. I have taken your request into consideration in the last commit.

bgschiller reviewed (commented):
Thanks for your changes!

There's a couple small edits I'd like to see. Also, could you remove the `# Modifed by mhoangvslev` and `# https://bitbukcet.org/astanin/python-tabulate` comments? I don't think they contribute to the clarity of the code.

I am happy to add you to a Contributors section in the README, though.

mhoangvslev committed:
Clean up per request from bgschiller

Indentation and clean ups

mhoangvslev committed:
Take into consideration console users

Re-introduce plain text support for console users

mhoangvslev committed:
Solution for issue #19

The kernel now catch any error and warn the user.
Also, all messages transmitted to client session will be displayed in result cell
